penumbra-zone/penumbra
# Allow building client crates without rocksdb

This required:
- feature gating `cnidarium` in `penumbra-app`
- removing the stateless transaction check in the view crate
- removing the ceremony contribution command from `pcli`

Closes #4885

## Checklist before requesting a review


  > Only client code was changed, the non-client code that was changed was only changed to have the public API be reduced without certain default features enabled. PD should be completely unchanged by this fix.
  
## How to test this:

run `cargo build --bin pcli -p pcli` (and pclientd, and pindexer) and observe that rocksdb is not built.
